homeassistant/components/mqtt/client.py

```diff
@@ -893,25 +893,47 @@ async def _async_perform_subscriptions(self) -> None:
         if not self._pending_subscriptions:
             return
 
-        subscriptions: dict[str, int] = self._pending_subscriptions
+        pending_subscriptions: dict[str, int] = self._pending_subscriptions
+        pending_wildcard_subscriptions = {
+            subscription.topic: pending_subscriptions.pop(subscription.topic)
+            for subscription in self._wildcard_subscriptions
+            if subscription.topic in pending_subscriptions
+        }
+        simple_subscription_list = list(pending_subscriptions.items())
```

Comment:
Why do we need to separate the wildcard subscriptions from other subscriptions?

Reply:
To chunk the subscription burst we need to pick the wildcard subscriptions first (we have 2 per platform) and process those. The whole idea of the PR is to leverage the load.

Reply:
The idea is that we get control on incoming load of replayed payloads. We used to subscribe to a wildcard causing all retained config to come in at once. We get more control on the load if we subscribe per platform. As we do that we won't get all subscriptions for all platforms at once.
Wildcard subscriptions we do one by one, normal subscriptions we perform in chunks as this is performance wise faster.
To able to do that, we need to split wildcard and normal subscriptions.